### PR TITLE
[HYDRA-398] - Show plugin properties in pipelines using old artifacts

### PR DESCRIPTION
**Pull request by ajainarayanan**
#### TL;DR
- Use artifact information from the pipeline once published. UI should not use the CDAP version.
#### Longer version
- UI is currently using CDAP version all the time to query plugin properties on a published pipeline. 
- This might be ok if we are querying for a plugin version that the current CDAP version supports. - - But if we want to open a plugin that was created in older versions in a newer version of Hydrator, we should use the artifact information in the pipeline instead of the CDAP version.

Bamboo build: http://builds.cask.co/browse/CDAP-DRC4220/latest
#### EDIT:
- Have also added fix for handling empty input schemas
- Have fixed every reference of `JSON.parse` to be wrapped in a `try...catch` block

### CODE REVIEW

**Comment by elfenheart** [comment]
👍  LGTM

File: cdap-ui/app/directives/complex-schema/complex-schema.js [hunk 1]
```diff
@@ -68,13 +69,26 @@ function ComplexSchemaController (avsc, SCHEMA_TYPES, $scope, uuid, $timeout, Sc
...
 
   function init(strJson) {
-    if (!strJson || strJson === 'record') {
-      recordName = vm.recordName || 'a' + uuid.v4().split('-').join('');
+    const isEmptySchema = (schemaJson) => {
+      if (!schemaJson) {
+        return true;
+      }
+      // FIXME: The weird _fields was encountered when I used imported through plugin-functions. 
+      if (angular.isObject(schemaJson) && !(schemaJson.fields||schemaJson._fields).length) {
```
Comment (elfenheart) [inline cdap-ui/app/directives/complex-schema/complex-schema.js]: _fields is internal avsc. You should do schemaJson.getFields()

**Comment by elfenheart** [comment]
👍  LGTM

**Comment by ajainarayanan** [comment]
Demoed it to @bdmogal
